pennylane/math/single_dispatch.py

```diff
@@ -263,9 +262,14 @@ def _take_autograd(tensor, indices, axis=None):
 ar.autoray._FUNC_ALIASES["tensorflow", "arccos"] = "acos"
 ar.autoray._FUNC_ALIASES["tensorflow", "arctan"] = "atan"
 ar.autoray._FUNC_ALIASES["tensorflow", "arctan2"] = "atan2"
-ar.autoray._FUNC_ALIASES["tensorflow", "diag"] = "diag"
 
 
+def _coerce_tensorflow_diag(x, **kwargs):
+    return ar.autoray.tensorflow_diag(_tf_convert_to_tensor(x), **kwargs)
+
+
+ar.register_function("tensorflow", "diag", _coerce_tensorflow_diag)

```

Comment:
The [recent changes to `diag` for Tensorflow in Autoray](https://github.com/jcmgray/autoray/commit/2c7704922fc4ef8d10ec96da98b6172d9fcd54d5) implemented a custom method, which is not compatible with the convention in PennyLane that we use `fn(x, like="tensorflow")` on a numpy array, for example. 
This is a PL-related issue because Autoray actually does not suggest that `like` should be used for functions that dispatch on the input, but [just for functions that need the interface as _additional_ (rather than overriding) information](https://github.com/jcmgray/autoray/blob/3b8efff7a10965453feb95e34dbb12299bedcf57/autoray/autoray.py#L73) :sweat_smile: